qiskit_experiments/characterization/spectroscopy.py

```diff
@@ -0,0 +1,408 @@
+        - sigma: The standard deviation of the pulse, the default is duration / 4.
+        - width: The width of the flat-top in the pulse, the default is 0, i.e. a Gaussian.
+
+        Args:
+            qubit: The qubit on which to run spectroscopy.
+            frequencies: The frequencies to scan in the experiment.
+            unit: The unit in which the user specifies the frequencies. Can be one
+                of 'Hz', 'kHz', 'MHz', 'GHz'. Internally, all frequencies will be converted
+                to 'Hz'.
+            absolute: Boolean to specify if the frequencies are absolute or relative to the
+                qubit frequency in the backend.
+
+        Raises:
+            QiskitError: if there are less than three frequency shifts or if the unit is not known.
+
+        """
+        if len(frequencies) < 3:
+            raise QiskitError("Spectroscopy requires at least three frequencies.")
+
+        if unit not in self.__units__:
+            raise QiskitError(f"Unsupported unit: {unit}.")
+
+        self._frequencies = [freq * self.__units__[unit] for freq in frequencies]
+        self._absolute = absolute
+
+        super().__init__([qubit], circuit_options=("amp", "duration", "sigma", "width"))
+
+    def circuits(self, backend: Optional["Backend"] = None, **circuit_options):
+        """Create the circuit for the spectroscopy experiment.
+
+        The circuits are based on a GaussianSquare pulse and a frequency_shift instruction
+        encapsulated in a gate.
+
+        Args:
+            backend: A backend object.
+            circuit_options: Key word arguments to run the circuits. The circuit options are
+                - amp: The amplitude of the GaussianSquare pulse, defaults to 0.1.
+                - duration: The duration of the GaussianSquare pulse, defaults to 1024 samples.
+                - sigma: The standard deviation of the GaussianSquare pulse, defaults to one
+                    fifth of the duration.
+                - width: The width of the flat top in the GaussianSquare pulse, defaults to 0.
+
+        Returns:
+            circuits: The circuits that will run the spectroscopy experiment.
+
+        Raises:
+            QiskitError: If relative frequencies are used but no backend was given.
+        """
+        if not backend and not self._absolute:
+            raise QiskitError("Cannot run spectroscopy relative to qubit without a backend.")
+
+        amp = circuit_options.get("amp", 0.1)
+        duration = circuit_options.get("duration", 1024)
+        sigma = circuit_options.get("sigma", duration / 4)
+        width = circuit_options.get("width", 0)
+
+        drive = pulse.DriveChannel(self.physical_qubits[0])
+
+        # Create a template circuit
+        freq_param = Parameter("frequency")
+        with pulse.build(name="spectroscopy") as sched:
+            pulse.set_frequency(freq_param, drive)
+            pulse.play(pulse.GaussianSquare(duration, amp, sigma, width), drive)
```

Comment:
Can you give the parameters as kwargs? This might be sensitive to change like this https://github.com/Qiskit/qiskit-terra/issues/6395

Reply:
Done here: c7de4c1